src/components/TimeCardPage/CellTypes/CommentCell.tsx

```diff
@@ -1,43 +1,196 @@
-import moment from 'moment-timezone';
-import React, {useState, useEffect} from 'react'; 
+import React, {useState, useEffect, useContext} from 'react'; 
+import {Button} from '@chakra-ui/react';
+import { UserContext } from '../UserContext';
+import {
+    Modal,
+    ModalOverlay,
+    ModalContent,
+    ModalHeader,
+    ModalFooter,
+    ModalBody,
+    ModalCloseButton,
+    Input,
+    IconButton,
+    ButtonGroup,
+    Flex,
+    useDisclosure,
+    useEditableControls
+  } from '@chakra-ui/react';
+import {
+    ChatIcon,
+    WarningIcon,
+    CheckIcon,
+    CloseIcon,
+    EditIcon
+  } from '@chakra-ui/icons';
+import {
+  Editable,
+  EditableTextarea,
+  EditablePreview,
+  EditableInput
+  } from '@chakra-ui/react';
 
 import {CommentSchema} from '../../../schemas/RowSchema'; 
-import {CommentType} from '../types'; 
-import {TIMEZONE} from '../../../constants'; 
+import {CommentType, CellStatus} from '../types'; 
+import CommentModal from '../CommentModal'
 
 interface CommentProps {
     comments: CommentSchema[] | undefined; 
     setComment: Function; 
 } 
 
+
+// setComments can be used for comments or reports
+interface ShowCommentModalProps{
+    comments: CommentSchema[];
+    setComments: Function;
+    icon;
+    color: string;
+    editable: boolean;
+}
+
+export const createNewComment = (type: CommentType, content: string) => {
+    return {
+      AuthorID:"david", 
+      Type: type, 
+      Timestamp: 1312313, 
+      Content: content, 
+      State: CellStatus.Active
+    }
+}
+
+function ShowCommentModal(props:ShowCommentModalProps) {
```

Comment:
Can destructure the props for easy access later:
```suggestion
function ShowCommentModal({
  comments,
  setComments,
  icon,
  color,
  editable
}: ShowCommentModalProps) {
```

then
```tsx
const [comment, setComment] = useState(props.comments[props.comments.length - 1]);
```
 becomes 
```tsx 
const [comment, setComment] = useState(comments[comments.length - 1]);
```